### ISSUE TITLE
Harry Potter 2 cos support

### ISSUE THREAD

**Issue by kevincrans** [xtremexp/UT4X-Converter#79]
since uc2 and dnf"1 are supported, which coincidentally have all binaries available for the same reason as the hp2 prototype, can support for harry potter 2 be added as well?
The Hp modding community does a lot work on those games and made it 64-bit.
There's a lot of people wanting to experiment with it in UE4, it can already be done using t3d and a supported "donor game".
The community's github: https://github.com/metallicafan212/HarryPotterUnrealWiki/wiki/

**Comment by xtremexp** [comment]
As stated before main goal of this program is to focus on UT series game (despite DNF was an exception).
So no support for HP2 is planned. Also the project is now in "passive mode".
However i had started some changes so people could try with other games.
Had planned to do ui for this  but not finished.

1) If HP2 have a package extractor (like /System/ucc.exe in UT99) you can try add a new entry
and  edit the ApplicationConfiguration.json in Documents/UT4X-Converter folder :

> {
>     "name" : "Harry Potter 2",
>     "ueVersion" : 2,
>     "mapFolder" : "/Maps",
>     "mapExt" : "<map_extension>",
>     "texExt" : "<tex_extension>",
>     "musicExt" : "<music_extension>",
>     "pkgExtractorPath" : "/System/ucc.com",
>     "path" : "C:\\Program Files (x86)\\Steam\\steamapps\\common\\Harry Potter 2",
>     "suggestedPath" : "C:\\Program Files (x86)\\Steam\\steamapps\\common\\Harry Potter 2?",
>     "useTexDb" : false,
>     "isCustom" : true,
>     "soundExt" : "<sound_extension>",
>     "pkgExtractor" : "umodel",
>     "disabled" : false,
>     "id" : "HP2",
>     "convertsToGames" : [ {
>       "gameId" : "UT4",
>       "scale" : 1.25
>     } ]
>   }

It is important "isCustom" to be set to true else if not it will be deleted at start of program.
You will have to replace with the good map extension, music extension, and other properties (i do not have HP2)

2) Just set HP2 folder as UT99 folder and convert only from .t3d file and uncheck convert textures/sounds/staticmeshes in conversion settings.

**Comment by kevincrans** [comment]
Ah right, thanks! Since the prototype has the source, it can also compile ucc.

**Comment by xtremexp** [comment]
I've just downloaded game from https://archive.org/details/dev-hpch . It has ucc.exe, it might work.
However many actors are new like "Ambient_light","Elevator_mover", ... does not exist in unreal game series
so in current version, they won't be converted.
I've quickly added support for these new actors: candleflame, Darklight, FogLight, SpecialLit, ThunderLightning, Torch_light, ElevatorMover, GridMover, GridMover2 . (note: haven't tested, just give a try)
I've commited the change. Just build and run program with the changes (or build the installer and install the new version).
.
In some future version, some day, actor match will be configurable.
